develop/tutorials/articles/210-application-security/04-authentication-pipelines/01-auto-login.markdown

```diff
@@ -0,0 +1,90 @@
+# Auto Login
+
+While @product@ supports a wide variety of [authentication mechanisms](/discover/deployment/-/knowledge_base/7-0/liferay-portal-security), 
+you may use a home-grown system or some other product to authenticate users. If
+this is your situation, you can write an Auto Login component to support your
+authentication system. 
+
+Auto Login components can check if the request contains something (a cookie, an
+attribute) that can be associated with a user in any way. If the component can
+make that association, it can authenticate that user to @product@. 
+
+## Creating an Auto Login Component
+
+Create a [Declarative Services component](/develop/tutorials/-/knowledge_base/7-0/creating-modules-with-liferay-ide#creating-component-classes). 
```

Comment:
We've had a lot of discussions with @JorgeFerrer about this. We do mention in the introductory materials that there are many ways to register a service, but the most straightforward way is to use Declarative Services, and so we "default" to that. I also wouldn't want to get lost in the weeds of an OSGi discussion when the tutorial is on creating an Auto Login component, and the example uses Declarative Services. I mostly did it this way just to provide a link to one of our other tutorials. If you feel strongly about this, I can revise it so it says "create a component."